packages/analytics/amplify_analytics_pinpoint/lib/amplify_analytics_pinpoint.dart

```diff
@@ -16,4 +16,12 @@
 /// Amplify Analytics Pinpoint
 library amplify_analytics_pinpoint;
 
+/// Overridable Flutter injected dependencies
+export 'package:amplify_analytics_pinpoint_dart/amplify_analytics_pinpoint_dart.dart'
+    show
+        AppLifecycleProvider,
+        CachedEventsPathProvider,
+        DeviceContextInfoProvider;
```

Comment:
I don't know if we should allow overriding these in dev-preview. Let's discuss since it will change the public API.

Reply:
Sure I can bring this up in our team meeting to see what consensus we get.  

Not sure how to run integration tests for AppLifecycleProvider if it's not exposed though.